### DISCUSSION TITLE
Can I assign values to json object using json::iterator?

### DISCUSSION THREAD

**Discussion by tr4v3ler** [nlohmann/json#4062] [General]
I wrote the following code, but an error was reported:
```
for (auto it = syscalls.begin(); it != syscalls.end(); it++) {
    json syscall = *it;
    if (syscall.contains("type")) {
        ...
    }
}
```
The type of syscalls is json::array.

**Comment by gregmarr** [comment]
First, this makes a copy of the object, so any changes you make to it won't be saved.

    json syscall = *it;

Second, you can't use `contains(string)` on an `array`, it only works on `object`.

**Comment by tr4v3ler** [comment]
> First, this makes a copy of the object, so any changes you make to it won't be saved.
> 
> ```
> json syscall = *it;
> ```
> 
> Second, you can't use `contains(string)` on an `array`, it only works on `object`.

Thanks for your reply. I try to save changes by this way:
```
*it = syscall;
```
It's ok. After a period of debugging, It's solved.
